src/transformers/utils/import_utils.py

```diff
@@ -642,6 +642,29 @@ def is_torch_mlu_available(check_device=False):
     return hasattr(torch, "mlu") and torch.mlu.is_available()
 
 
+@lru_cache()
+def is_torch_musa_available(check_device=False):
+    "Checks if `torch_musa` is installed and potentially if a MUSA is in the environment"
+    if not _torch_available or importlib.util.find_spec("torch_musa") is None:
+        return False
+
+    import torch
+    import torch_musa  # noqa: F401
+
+    from ..dependency_versions_table import deps
+
+    deps["accelerate"] = "accelerate>=0.33.0"
```

Comment:
I think this is OK given the existing logic for mlu

Reply:
Yep error is best then indeed. Wanted to make sure that’s the way to do this before nit picking!

(intention is raise an err if we don’t have that version or higher accelerate)